tests/test_psf.py

```diff
@@ -0,0 +1,278 @@
+
+    return (np.exp(-0.5*(u**2 + (v*ab)**2)/sigma1**2)
+            + b*np.exp(-0.5*(u**2 + (v*ab)**2)/sigma2**2))/(1 + b)
+
+
+class SpatialModelPsfTestCase(lsst.utils.tests.TestCase):
+    """A test case for SpatialModelPsf"""
+
+    def measure(self, footprintSet, exposure):
+        """Measure a set of Footprints, returning a SourceCatalog"""
+        catalog = afwTable.SourceCatalog(self.schema)
+
+        footprintSet.makeSources(catalog)
+
+        self.measureSources.run(catalog, exposure)
+        return catalog
+
+    def setUp(self):
+        config = SingleFrameMeasurementTask.ConfigClass()
+        config.slots.apFlux = 'base_CircularApertureFlux_12_0'
+        self.schema = afwTable.SourceTable.makeMinimalSchema()
+
+        self.measureSources = SingleFrameMeasurementTask(
+            self.schema, config=config
+        )
+        self.usePsfFlag = self.schema.addField("use_psf", type="Flag")
+
+        width, height = 110, 301
+
+        self.mi = afwImage.MaskedImageF(geom.ExtentI(width, height))
+        self.mi.set(0)
+        sd = 3  # standard deviation of image
+        self.mi.getVariance().set(sd*sd)
+        self.mi.getMask().addMaskPlane("DETECTED")
+
+        self.ksize = 31  # size of desired kernel
+
+        sigma1 = 1.75
+        sigma2 = 2*sigma1
+
+        self.exposure = afwImage.makeExposure(self.mi)
+        self.exposure.setPsf(measAlg.DoubleGaussianPsf(self.ksize, self.ksize,
+                                                       1.5*sigma1, 1, 0.1))
+        cdMatrix = np.array([1.0, 0.0, 0.0, 1.0])
+        cdMatrix.shape = (2, 2)
+        wcs = afwGeom.makeSkyWcs(crpix=geom.PointD(0, 0),
+                                 crval=geom.SpherePoint(0.0, 0.0, geom.degrees),
+                                 cdMatrix=cdMatrix)
+        self.exposure.setWcs(wcs)
+
+        #
+        # Make a kernel with the exactly correct basis functions.
+        # Useful for debugging
+        #
+        basisKernelList = []
+        for sigma in (sigma1, sigma2):
+            basisKernel = afwMath.AnalyticKernel(
+                self.ksize, self.ksize, afwMath.GaussianFunction2D(sigma, sigma)
+            )
+            basisImage = afwImage.ImageD(basisKernel.getDimensions())
+            basisKernel.computeImage(basisImage, True)
+            basisImage /= np.sum(basisImage.getArray())
+
+            if sigma == sigma1:
+                basisImage0 = basisImage
+            else:
+                basisImage -= basisImage0
+
+            basisKernelList.append(afwMath.FixedKernel(basisImage))
+
+        order = 1  # 1 => up to linear
+        spFunc = afwMath.PolynomialFunction2D(order)
+
+        exactKernel = afwMath.LinearCombinationKernel(basisKernelList, spFunc)
+        exactKernel.setSpatialParameters(
+            [[1.0, 0, 0],
+             [0.0, 0.5*1e-2, 0.2e-2]]
+        )
+
+        rand = afwMath.Random()  # make these tests repeatable by setting seed
+
+        addNoise = True
+
+        if addNoise:
```

Comment:
I don't understand the point of those lines; can't we just get rid of the if statement altogether?

(Again, I know this is from meas_extensions_psfex!)

Reply:
Haha.  Yes, let's get rid of them.  Cargo culting at its best... :)